### PR TITLE
Adds a new map to the game, Blueshift! Created by IsaacTheSharkWolf.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request
dreamseeker_4CpttOZUnW
dreamseeker_AlRF8YlZCs
dreamseeker_QMQGRS0bEM
dreamseeker_kFUUTbdmFT
dreamseeker_3n87jlbup3
dreamseeker_syIC2hukZc
dreamseeker_7sFWbbrcYx
### Floor 1
StrongDMM_JIaLaiMk10
### Floor 2
StrongDMM_KFtGpZl09x
### Floor 3
StrongDMM_Cib7r8GLH7

## Why It's Good For The Game

It's a very cool map. I like it a lot. And we need to do some debugging.

## Changelog

:cl: IsaacTheSharkWolf
add: Adds a new map to the game, Blueshift!
/:cl:

### CODE REVIEW

**Comment by SuperNovaa41** [comment]
Could you post an overview of entire map?

**Comment by Cheshify** [comment]
GOOF SHOW THE FULL MAP

**Comment by poobfarder** [comment]
this is the skyrat map, goofball

**Comment by capsaicinz** [comment]
knowing this is from skyrat, a very very high pop server compared to tg, it seems kind of

big

and also pretty laggy at times from my experience

**Comment by RandomGamer123** [comment]
Triple z-levels but yeah this'll have to be for like 60+ pop or at least 40+ not 25+

**Comment by Iamgoofball** [comment]
>and also pretty laggy at times from my experience

This is part of what we're trying to debug here.

File: _maps/_basemap.dm [hunk 1]
```diff
@@ -12,7 +12,9 @@
 		#include "map_files\MetaStation\MetaStation.dmm"
 		#include "map_files\IceBoxStation\IceBoxStation.dmm"
 		#include "map_files\tramstation\tramstation.dmm"
-
+		#include "map_files\Blueshift\BlueShift_lower.dmm"
```
Comment (JohnFulpWillard) [inline _maps/_basemap.dm:15]: Are these all supposed to be defined here? It isn't like this for Icebox.

File: code/modules/shuttle/docking.dm [hunk 2]
```diff
@@ -173,6 +173,12 @@
 		var/turf/oldT = old_turfs[i]
 		var/turf/newT = new_turfs[i]
 		newT.afterShuttleMove(oldT, rotation) //turfs
+		var/turf/newCeiling = get_step_multiz(newT, UP)
```
Comment (JohnFulpWillard) [inline code/modules/shuttle/docking.dm:176]: can you make this in a separate PR please it would be nice to have :)

File: _maps/_basemap.dm (hunk 1 above)
Comment (Iamgoofball) [reply _maps/_basemap.dm:15]: i believe ice box's separate levels not being here is an oversight

Comment (Iamgoofball) [reply _maps/_basemap.dm:15]: as in, we used to have the z-levels on the icebox dmm itself but we split them recently without updating it here

Comment (JohnFulpWillard) [reply _maps/_basemap.dm:15]: it isnt in the .dmm  but I believe is in the .json

**Comment by optimumtact** [comment]
why doesn't skyrat get some devs with a clue and not fob their failures to write good code off on us.

**Comment by IsaacTheSharkWolf** [comment]
Just a quick note, this is a very barebones conversion, there are gaps of missing content and i haven't tidied up the map a lot, from my understanding it's to test the map and see if it runs into the same performance issues here.
I'm ofcourse curious to the reception of my map and if you guys want to actually see it in rotation i can always make a proper conversion.

File: _maps/_basemap.dm (hunk 1 above)
Comment (Iamgoofball) [reply _maps/_basemap.dm:15]: gamer, this is the "compile all maps" CI task
every dmm file should be in there

**Comment by IsaacTheSharkWolf** [comment]
in fact this is an unreleased half finished update build, this isn't currently in rotation on skyrat.

**Comment by SuperNovaa41** [comment]
> Just a quick note, this is a very barebones conversion, there are gaps of missing content and i haven't tidied up the map a lot, from my understanding it's to test the map and see if it runs into the same performance issues here. I'm ofcourse curious to the reception of my map and if you guys want to actually see it in rotation i can always make a proper conversion.

I think it looks really cool but it's far too large for our average player base size

**Comment by optimumtact** [comment]
we would never want this map in rotation because it's too big which is why it lags, which is why this PR is stupid waste of time in the first place.

**Comment by EOBGames** [comment]
To second oranges, this would not be a permanent addition to our codebase in this form- we have different map standards to Skyrat that would require being met, and constraints on our number of maps such that this would require replacing another map. I'm not sure why Goof has attempted to push this as such, as he didn't pitch it as such to me. The way he explained it sounded like he wanted a temporary testmerge to determine if lag was codebase agnostic or codebase specific to Skyrat. To be clear: with the correct discussion beforehand, I wouldn't necessarily be against this map, but that discussion hasn't been had. For testing the map on our servers to see if it suffers the same issues as downstream, this PR suffices since closed PRs can still be testmerged (and event rounds run accordingly).

As a quick aside, since this is the first station PR we've had in a while (and proper protocol wasn't followed on it), I'd like to take this opportunity to remind everyone of our policy regarding station map addition/removal:
> All pull requests adding, removing, or replacing station maps must receive prior approval from a maptainer, or they will be closed without additional warning.

This does mean explicit approval, i.e. a direct yes, and preferably it should come from me (not to place myself over my colleagues as the "head maptainer", but I try to reserve executive control over additions and removals to maintain a firm guiding hand on the map pool).
